plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/TrinoJdbcCatalog.java

```diff
@@ -0,0 +1,348 @@
+import static java.util.Objects.requireNonNull;
+import static org.apache.iceberg.CatalogUtil.dropTableData;
+
+public class TrinoJdbcCatalog
+        extends AbstractTrinoCatalog
+{
+    private final JdbcCatalog jdbcCatalog;
+    private final TrinoFileSystemFactory fileSystemFactory;
+    private final String defaultWarehouseDir;
+    private final Map<SchemaTableName, TableMetadata> tableMetadataCache = new ConcurrentHashMap<>();
+
+    public TrinoJdbcCatalog(
+            CatalogName catalogName,
+            TypeManager typeManager,
+            IcebergTableOperationsProvider tableOperationsProvider,
+            JdbcCatalog jdbcCatalog,
+            TrinoFileSystemFactory fileSystemFactory,
+            boolean useUniqueTableLocation,
+            String defaultWarehouseDir)
+    {
+        super(catalogName, typeManager, tableOperationsProvider, useUniqueTableLocation);
+        this.jdbcCatalog = requireNonNull(jdbcCatalog, "jdbcCatalog is null");
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
+        this.defaultWarehouseDir = requireNonNull(defaultWarehouseDir, "defaultWarehouseDir is null");
+    }
+
+    @Override
+    public boolean namespaceExists(ConnectorSession session, String namespace)
+    {
+        return jdbcCatalog.namespaceExists(Namespace.of(namespace));
+    }
+
+    @Override
+    public List<String> listNamespaces(ConnectorSession session)
+    {
+        return jdbcCatalog.listNamespaces().stream()
+                .map(namespace -> namespace.level(0).toLowerCase(ENGLISH))
+                .collect(toImmutableList());
+    }
+
+    @Override
+    public Map<String, Object> loadNamespaceMetadata(ConnectorSession session, String namespace)
+    {
+        return jdbcCatalog.loadNamespaceMetadata(Namespace.of(namespace)).entrySet().stream()
+                .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    @Override
+    public Optional<TrinoPrincipal> getNamespacePrincipal(ConnectorSession session, String namespace)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public void createNamespace(ConnectorSession session, String namespace, Map<String, Object> properties, TrinoPrincipal owner)
+    {
+        try {
+            jdbcCatalog.createNamespace(Namespace.of(namespace), transformValues(properties, String.class::cast));
+        }
+        catch (RuntimeException e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR, "Failed to create a namespace " + namespace, e);
+        }
+    }
+
+    @Override
+    public void dropNamespace(ConnectorSession session, String namespace)
+    {
+        jdbcCatalog.dropNamespace(Namespace.of(namespace));
+    }
+
+    @Override
+    public void setNamespacePrincipal(ConnectorSession session, String namespace, TrinoPrincipal principal)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "setNamespacePrincipal is not supported for Iceberg JDBC catalog");
```

Comment:
Grammar nit pick, looks like a copy from the REST wording
```suggestion
        throw new TrinoException(NOT_SUPPORTED, "setNamespacePrincipal is not supported for Iceberg JDBC catalogs");
```